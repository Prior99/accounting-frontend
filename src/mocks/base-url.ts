declare var global: any;
global.baseUrl = "example.com";
