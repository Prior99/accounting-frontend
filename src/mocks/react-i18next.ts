import * as React from "react";

jest.mock("react-i18next", () => ({
    translate: (namespaces?: string[], options?: any) => <C extends Function>(WrappedComponent: C) => {
        class Translated extends React.Component<{}, undefined> {
            public render() {
                const props = Object.assign({}, this.props, {
                    t: (name: string) => name,
                });
                return React.createElement(WrappedComponent as any, props);
            }
        }
        return Translated;
    },
}));
