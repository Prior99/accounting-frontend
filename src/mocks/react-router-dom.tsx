import * as React from "react";

jest.mock("react-router-dom", () => ({
    Link: ({ to, children }: { to: string, children: JSX.Element }) => (<a href={to}>{children}</a>),
}));
