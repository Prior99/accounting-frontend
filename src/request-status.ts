export enum RequestStatus {
    PENDING = "pending",
    IN_PROGRESS = "in_progress",
    FAIL = "fail",
    SUCCESS = "success",
}
