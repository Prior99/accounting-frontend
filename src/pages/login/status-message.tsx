import * as React from "react";
import { translate, InjectedTranslateProps } from "react-i18next";
import { RequestStatus } from "request-status";
import { Message } from "semantic-ui-react";

@translate(["login", "common"])
export class StatusMessage extends React.Component<{ status: RequestStatus } & InjectedTranslateProps, undefined> {
    public render() {
        const { status, t } = this.props;
        switch (status) {
            case RequestStatus.FAIL:
                return (
                    <Message
                        warning
                        icon="warning sign"
                        header={t("loginFailed.headline")}
                        content={t("loginFailed.content")}
                    />
                );
            default:
                return null; // tslint:disable-line
        }
    }
}
