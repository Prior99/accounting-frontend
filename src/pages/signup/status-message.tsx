import * as React from "react";
import { translate, InjectedTranslateProps } from "react-i18next";
import { RequestStatus } from "request-status";
import { Message, Icon } from "semantic-ui-react";

@translate(["login", "common"])
export class StatusMessage extends React.Component<{ status: RequestStatus } & InjectedTranslateProps, undefined> {
    public render() {
        const { status, t } = this.props;
        switch (status) {
            case RequestStatus.SUCCESS:
                return (
                    <Message
                        success
                        icon="checkmark"
                        header={t("signupSuccess.headline")}
                        content={t("signupSuccess.content")}
                    />
                );
            case RequestStatus.FAIL:
                return (
                    <Message
                        warning
                        icon="warning sign"
                        header={t("signupFailed.headline")}
                        content={t("signupFailed.content")}
                    />
                );
            case RequestStatus.IN_PROGRESS:
                return (
                    <Message success icon>
                        <Icon name="circle notched" loading />
                        <Message.Content>
                            <Message.Header>
                                {t("inProgress.headline")}
                            </Message.Header>
                            {t("inProgress.content")}
                        </Message.Content>
                    </Message>
                );
            default:
                return null; // tslint:disable-line
        }
    }
}
