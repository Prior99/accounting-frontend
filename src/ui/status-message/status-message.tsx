import * as React from "react";
import { RequestStatus } from "request-status";
import { Message, Icon } from "semantic-ui-react";

type Props = {
    readonly status: RequestStatus;
    readonly successHeadline?: string;
    readonly successContent?: string;
    readonly failHeadline?: string;
    readonly failContent?: string;
    readonly inProgressHeadline?: string;
    readonly inProgressContent?: string;
};

export class StatusMessage extends React.Component<Props, undefined> {
    public render() {
        const {
            status,
            successHeadline,
            successContent,
            failHeadline,
            failContent,
            inProgressHeadline,
            inProgressContent,
        } = this.props;
        switch (status) {
            case RequestStatus.SUCCESS:
                return (
                    <Message
                        success
                        icon="checkmark"
                        header={successHeadline}
                        content={successContent}
                    />
                );
            case RequestStatus.FAIL:
                return (
                    <Message
                        warning
                        icon="warning sign"
                        header={failHeadline}
                        content={failContent}
                    />
                );
            case RequestStatus.IN_PROGRESS:
                return (
                    <Message success icon>
                        <Icon name="circle notched" loading />
                        <Message.Content>
                            {
                                typeof inProgressHeadline === "undefined" &&
                                <Message.Header>
                                    {inProgressHeadline}
                                </Message.Header>
                            }
                            {inProgressContent}
                        </Message.Content>
                    </Message>
                );
            default:
                return null; // tslint:disable-line
        }
    }
}
