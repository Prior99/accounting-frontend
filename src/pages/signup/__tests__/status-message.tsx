import "mocks/react-i18next";

import * as React from "react";
import { StatusMessage } from "../status-message";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import { RequestStatus } from "request-status";

[RequestStatus.PENDING, RequestStatus.FAIL, RequestStatus.SUCCESS, RequestStatus.IN_PROGRESS].forEach(status => {
    test("`StatusMessage` is rendered correctly with the status being `PENDING`", () => {
        const mounted = mount(<StatusMessage status={status} />);
        expect(toJson(mounted)).toMatchSnapshot();
    });
});
