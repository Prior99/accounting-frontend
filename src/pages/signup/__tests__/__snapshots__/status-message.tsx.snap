// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`StatusMessage\` is rendered correctly with the status being \`PENDING\` 1`] = `
<Translated
  status="pending"
>
  <StatusMessage
    status="pending"
    t={[Function]}
  />
</Translated>
`;

exports[`\`StatusMessage\` is rendered correctly with the status being \`PENDING\` 2`] = `
<Translated
  status="fail"
>
  <StatusMessage
    status="fail"
    t={[Function]}
  >
    <Message
      content="signupFailed.content"
      header="signupFailed.headline"
      icon="warning sign"
      warning={true}
    >
      <div
        className="ui icon warning message"
      >
        <Icon
          as="i"
          name="warning sign"
        >
          <i
            aria-hidden="true"
            className="warning sign icon"
          />
        </Icon>
        <MessageContent>
          <div
            className="content"
          >
            <MessageHeader
              content="signupFailed.headline"
            >
              <div
                className="header"
              >
                signupFailed.headline
              </div>
            </MessageHeader>
            <p>
              signupFailed.content
            </p>
          </div>
        </MessageContent>
      </div>
    </Message>
  </StatusMessage>
</Translated>
`;

exports[`\`StatusMessage\` is rendered correctly with the status being \`PENDING\` 3`] = `
<Translated
  status="success"
>
  <StatusMessage
    status="success"
    t={[Function]}
  >
    <Message
      content="signupSuccess.content"
      header="signupSuccess.headline"
      icon="checkmark"
      success={true}
    >
      <div
        className="ui icon success message"
      >
        <Icon
          as="i"
          name="checkmark"
        >
          <i
            aria-hidden="true"
            className="checkmark icon"
          />
        </Icon>
        <MessageContent>
          <div
            className="content"
          >
            <MessageHeader
              content="signupSuccess.headline"
            >
              <div
                className="header"
              >
                signupSuccess.headline
              </div>
            </MessageHeader>
            <p>
              signupSuccess.content
            </p>
          </div>
        </MessageContent>
      </div>
    </Message>
  </StatusMessage>
</Translated>
`;

exports[`\`StatusMessage\` is rendered correctly with the status being \`PENDING\` 4`] = `
<Translated
  status="in_progress"
>
  <StatusMessage
    status="in_progress"
    t={[Function]}
  >
    <Message
      icon={true}
      success={true}
    >
      <div
        className="ui icon success message"
      >
        <Icon
          as="i"
          loading={true}
          name="circle notched"
        >
          <i
            aria-hidden="true"
            className="circle notched loading icon"
          />
        </Icon>
        <MessageContent>
          <div
            className="content"
          >
            <MessageHeader>
              <div
                className="header"
              >
                inProgress.headline
              </div>
            </MessageHeader>
            inProgress.content
          </div>
        </MessageContent>
      </div>
    </Message>
  </StatusMessage>
</Translated>
`;
